### PR TITLE
feat(dashboard): "Under Development" message

### PR DESCRIPTION
**Pull request by lucasbracher**
- shows "Under Development" message when "Devices" and "Labs" labels
   are clicked
 - labels change when clicked

Closes https://github.com/kernelci/dashboard/issues/223

### CODE REVIEW

File: dashboard/src/components/SideMenu/SideMenu.tsx [hunk 1]
```diff
@@ -20,13 +22,14 @@ import {
...
 type MenuItems = {
+  id: number;
   onClick: () => void;
   idIntl: MessagesKey;
   icon: JSX.Element;
   selected: boolean;
 };
 
-const emptyFunc = (): void => {};
+//const emptyFunc = (): void => {};
```
Comment (lfjnascimento) [inline dashboard/src/components/SideMenu/SideMenu.tsx:32]: leftover

File: dashboard/src/components/SideMenu/SideMenu.tsx [hunk 2]
```diff
@@ -68,19 +72,22 @@ const SideMenu = (): JSX.Element => {
 
   const items: MenuItems[] = [
     {
+      id: 0,
```
Comment (lfjnascimento) [inline dashboard/src/components/SideMenu/SideMenu.tsx:75]: we can just use the item index for this purpose

Comment (WilsonNet) [reply dashboard/src/components/SideMenu/SideMenu.tsx:83]: we should use the path url for this

File: dashboard/src/routes/underDevelopment.tsx [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+import { z } from 'zod';
+
+import { FormattedMessage } from 'react-intl';
+
+import { zOrigin } from '@/types/tree/Tree';
+
+export const HomeSearchSchema = z.object({
+  origin: zOrigin,
+});
+
```
Comment (WilsonNet) [inline dashboard/src/routes/underDevelopment.tsx:12]: the zOrigin is already in the root path, we don't need it here

File: dashboard/src/components/SideMenu/SideMenu.tsx [hunk 4]
```diff
@@ -49,6 +54,7 @@ const linkItems: LinkMenuItems[] = [
 const emptyFunc = (): void => {};
 
 const SideMenu = (): JSX.Element => {
+  const [selectedItemId, setSelectedItemId] = useState<number>(0);
```
Comment (WilsonNet) [inline dashboard/src/components/SideMenu/SideMenu.tsx:57]: we can extract the selected item just by getting the path from the router and seeing if equals the path that the link brings us

Comment (WilsonNet) [reply dashboard/src/components/SideMenu/SideMenu.tsx:57]: no need for useState here

File: dashboard/src/components/SideMenu/SideMenu.tsx [hunk 5]
```diff
@@ -126,10 +129,15 @@ const SideMenu = (): JSX.Element => {
...
+              item.id === selectedItemId
+                ? selectedItemClassName
+                : notSelectedItemClassName
             }
             key={item.idIntl}
-            onClick={item.onClick}
+            onClick={() => {
+              setSelectedItemId(item.id); // Atualizando o item selecionado
+              item.onClick(); // Navegando ao clicar
+            }}
```
Comment (WilsonNet) [inline dashboard/src/components/SideMenu/SideMenu.tsx:140]: Remove the portuguese comments also

File: dashboard/src/components/SideMenu/SideMenu.tsx [hunk 6]
```diff
@@ -46,47 +46,65 @@ const linkItems: LinkMenuItems[] = [
...
-  const useNavigateTo = (path: string): (() => void) => {
-    const router = useRouter();
-    const { origin: unsafeOrigin } = useSearch({ strict: false });
+  const { pathname } = useLocation();
+
+  const isCurrentPath = pathname === item.navigateTo;
+
+  const router = useRouter();
+
+  const { origin: unsafeOrigin } = useSearch({ strict: false });
```
Comment (lfjnascimento) [inline dashboard/src/components/SideMenu/SideMenu.tsx:65]: A pattern that helps us not to break the hook rules is to aggregate all hooks at the beginning of the component

**Pull request merged**